contrib/osx/keybindings.el

```diff
@@ -7,14 +7,17 @@
 
   ;; Keybindings
   (global-set-key (kbd "s-q") 'save-buffers-kill-terminal)
-  (global-set-key (kbd "s-v") 'yank)
-  (global-set-key (kbd "s-c") 'kill-ring-save)
+  (global-set-key (kbd "s-v") 'evil-paste-after)
+  (global-set-key (kbd "s-c") 'evil-yank)
+  (global-set-key (kbd "s-a") 'mark-whole-buffer)
   (global-set-key (kbd "s-x") 'kill-region)
   (global-set-key (kbd "s-w") 'kill-this-buffer)
-  (global-set-key (kbd "s-z") 'undo-tree-undo)
```

Comment:
This is interesting, reading the source it looks like `undo-tree-mode` doesn't redefine the `undo` function, it switches the keybindings to `undo-tree-undo'. However the default spacemacs `u` key is bound to just `undo`.

Reading the source of `undo-tree-undo` it looks like it it only makes a difference if you are in the undo-tree visualization. I think this is a bug in spacemacs and if you hit `u` while visiting a previous state in the undo tree it will wreck stuff.

Anyhow I think since Spacemacs defaults to `undo-tree-mode` it should do that in the keybindings, or call a special function that decides what to do based on if `undo-tree-mode` is enabled. You can always set them to the plain functions in your dotfile.